applications/StructuralMechanicsApplication/custom_processes/shell_to_solid_shell_process.cpp

```diff
@@ -0,0 +1,283 @@
+
+        IndexType node_id = total_number_of_nodes + i + 1;
+        NodeType::Pointer p_node0 = auxiliar_model_part.CreateNewNode(node_id, coordinates[0], coordinates[1], coordinates[2]);
+
+        // Set the DOFs in the nodes
+        for (auto it_dof = dofs.begin(); it_dof != dofs.end(); ++it_dof)
+            p_node0->pAddDof(*it_dof);
+
+        for (IndexType j = 0; j < number_of_layers; ++j) {
+            coordinates += normal * delta_thickness;
+            node_id = (j + 1) * geometry_number_of_nodes + total_number_of_nodes + i + 1;
+            NodeType::Pointer p_node1 = auxiliar_model_part.CreateNewNode(node_id, coordinates[0], coordinates[1], coordinates[2]);
+            // Set the DOFs in the nodes
+            for (auto it_dof = dofs.begin(); it_dof != dofs.end(); ++it_dof)
+                p_node1->pAddDof(*it_dof);
+        }
+
+        // We set the flag TO_ERASE for later remove the nodes
+        it_node->Set(TO_ERASE, true);
+    }
+
+    const std::string& element_name = mThisParameters["element_name"].GetString();
+    Element const& r_clone_element = KratosComponents<Element>::Get(element_name);
+    KRATOS_ERROR_IF_NOT(r_clone_element.GetGeometry().size() == 2 * TNumNodes) << "ERROR: Element " << element_name << " has a different number of nodes to " << 2 * TNumNodes << std::endl;
+
+    // We create the new elements
+    IndexType element_counter = total_number_of_elements;
+    for(IndexType i = 0; i < geometry_number_of_elements; ++i) {
+        auto it_elem = elements_array.begin() + i;
+
+        auto p_prop = it_elem->pGetProperties();
+        for (IndexType j = 0; j < number_of_layers; ++j) {
+            std::vector<IndexType> element_node_ids (2 * TNumNodes);
+            for (IndexType k = 0; k < TNumNodes; ++k) {
+                const IndexType index_node = it_elem->GetGeometry()[k].Id();
+                element_node_ids[k] = total_number_of_nodes + index_node + j * geometry_number_of_nodes;
+                element_node_ids[k + TNumNodes] = total_number_of_nodes + index_node + (j + 1) * geometry_number_of_nodes;
+            }
+            element_counter++;
+            auxiliar_model_part.CreateNewElement(element_name, element_counter, element_node_ids, p_prop);
+        }
+
+        // We set the flag TO_ERASE for later remove the elements
+        it_elem->Set(TO_ERASE, true);
+    }
+
+    // Finally we remove the old nodes and elements
+    mrThisModelPart.RemoveNodesFromAllLevels(TO_ERASE);
+    mrThisModelPart.RemoveElementsFromAllLevels(TO_ERASE);
+
+    // We copy the new model part to the original one
+    mrThisModelPart.AddNodes( auxiliar_model_part.NodesBegin(), auxiliar_model_part.NodesEnd() );
+    mrThisModelPart.AddElements( auxiliar_model_part.ElementsBegin(), auxiliar_model_part.ElementsEnd() );
+    
+    // Reorder again all the IDs
+    ReorderAllIds();
+
+//     // We initialize the new elements
+//     InitializeElements();
```

Comment:
why is this commented?
Is it needed or not?

Reply:
Depends where the process is called, maybe I could add the option, but the idea is to call it just after the model part is created (after reading the mdpa)